### PR TITLE
core: Add filtering by index

### LINKED ISSUE #673: Proposal: filter by model-scenario combinations
Through our discussion on slack, we identified a desire for a feature to allow for filtering unique model-scenario combinations. Here is one proposed implementation:

```
# data is a pyam.IamDataFrame
# ms_df if a dataframe with columns model and scenario

def subset_model_scenario(data, ms_df):
    idxs = [data.slice(model=m, scenario=s) for m, s in list(ms_df.itertuples(index=False, name=None))]
    idx = functools.reduce(operator.or_, idxs, False)
    return data[idx]
```

This utilizes the new `IamSlice` feature, constructing a list of slices of size `N` where `N` is the number of unique combinations and then using `logical_or` across all entries.

Thoughts @coroa @znicholls @danielhuppmann (and others)?

### PR DESCRIPTION
**Pull request by coroa**
Fixes #673 .

# Please confirm that this PR has done the following:

- [x] Tests Added
- [ ] Documentation Added
- [x] Name of contributors Added to AUTHORS.rst
- [ ] Description in RELEASE_NOTES.md Added

## Adding to RELEASE_NOTES.md (remove section after adding to RELEASE_NOTES.md)

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

# Description of PR

Tentative implementation for a filter kwarg supporting to select a list of pathways.

### CODE REVIEW

**Comment by coroa** [comment]
Failing tests are iiasa and unfcc connectors. Pbb unrelated to this PR.

**Comment by coroa** [comment]
> Failing tests are iiasa and unfcc connectors. Pbb unrelated to this PR.

Thanks for fixing, @danielhuppmann . Incorporated.

**Comment by gidden** [comment]
I've added minimal docstrings and to the release notes. @coroa this looks good to go from my side - good for you?

**Comment by gidden** [comment]
And any final thoughts from @danielhuppmann also welcome. Unless there's more to add, will merge when tests pass.

File: pyam/core.py [hunk 1]
```diff
@@ -1782,6 +1782,7 @@ def slice(self, keep=True, **kwargs):
          - 'meta' columns: filter by string value of that column
          - 'model', 'scenario', 'region', 'variable', 'unit':
            string or list of strings, where `*` can be used as a wildcard
+         - 'index': list of model, scenario 2-tuples
```
Comment (danielhuppmann) [inline pyam/core.py:1785]:
```suggestion
         - 'index': list of model, scenario 2-tuples or :class:`pandas.MultiIndex`
```

**Comment by gidden** [comment]
Thank you @coroa !

**Pull request merged**